### PR TITLE
Swik 2281 login stack exceed try2

### PR DESCRIPTION
**Pull request by TBoonX**
Fix number 2.

In order to test the social login error behaviour use the following code:
```
        this.handleDummyClick = this.handleDummyClick.bind(this);

    handleDummyClick(e) {
        e.preventDefault();
        this.provider = 'github';
        $('.ui.login.modal').modal('hide');
        localStorage.setItem(MODI, 'login');
        localStorage.setItem(NAME, '');
        let thatContext = this.context;
        async.series([
            function(callback) {
                thatContext.executeAction(newSocialData, {});
                callback(null, 1);
            },
            function(callback) {
                thatContext.executeAction(userSocialSignIn, {});
                callback(null, 2);
            }
        ]);
    }

                          <br/><br/>
                          <a href="#" onClick={this.handleDummyClick}>
                            dummy
                          </a>
```

### CODE REVIEW

File: components/Login/LoginModal.js [hunk 1]
```diff
@@ -167,14 +168,12 @@ class LoginModal extends React.Component {
             })
             .catch((action) => {
                 // console.log('action after click', action);
-                localStorage.setItem(MODI, 'login_failed');
-
                 //delete old data
                 let that = this;
                 async.series([
```
Comment (rmeissn) [inline components/Login/LoginModal.js]: Has not effect compared to:
```
that.context.executeAction(newSocialData, {});
if (action !== 'close')
    that.handleLoginButton();
```

File: components/Login/LoginModal.js [hunk 2]
```diff
@@ -319,11 +318,11 @@ class LoginModal extends React.Component {
         async.series([
```
Comment (rmeissn) [inline components/Login/LoginModal.js]: Has no effect compared to:
```
hatContext.executeAction(newSocialData, data);
thatContext.executeAction(userSocialSignIn, data);
```

File: components/Login/LoginModal.js [hunk 3]
```diff
@@ -134,8 +134,9 @@ class LoginModal extends React.Component {
     componentDidUpdate() {
         if (this.props.errorMessage.length > 2)
             $('.ui.form.signin').form('add errors', [this.props.errorMessage]);
-        console.log('componentDidUpdate:', this.props.errorMessage, ',', this.props.socialLoginError, ',', this.props.userid, ',', this.props.username, ',', this.state.isLoading);
+        // console.log('componentDidUpdate:', this.props.errorMessage, ',', this.props.socialLoginError, ',', this.props.userid, ',', this.props.username, ',', this.state.isLoading);
         if (localStorage.getItem(MODI) === 'login' && this.props.socialLoginError){
+            localStorage.setItem(MODI, 'login_failed');
```
Comment (rmeissn) [inline components/Login/LoginModal.js]: Is localstorage really needed for login?

Comment (TBoonX) [reply components/Login/LoginModal.js]: for social login yes. for normal login not. Could be improved later.

**Pull request merged**